src/format/strftime.rs

```diff
@@ -227,9 +314,16 @@ impl<'a> Iterator for StrftimeItems<'a> {
                 }
 
                 macro_rules! recons {
-                    [$head:expr, $($tail:expr),+] => ({
-                        const RECONS: &'static [Item<'static>] = &[$($tail),+];
-                        self.recons = RECONS;
+                    [$head:expr, $($tail:expr),+ $(,)*] => ({
+                        #[cfg(feature = "locales")]
+                        {
+                            self.recons = vec![$($tail),+];
```

Comment:
We shouldn't stick a `vec` call in here, it will cause a new vec to be allocated every iteration of the parse loop, it should be possible to do the same trick as below? Actually I'm not sure why this branching on whether the locales feature is enabled is required?